### ISSUE TITLE
Performance regression in the write test in 2.1-rc3

### ISSUE THREAD

**Issue by vladzcloudius** [scylladb/scylla#3250]
*Installation details*

Scylla version (or git commit hash): 3ab1c8abffdd2fe84945089ba0d363e357c65b55
Cluster size: 1
OS: Fedora 26 

*Hardware details*
* Scylla server:
  * Platform: physical
  * Architecture: x86_64
  * CPU: 
    * Model: Intel(R) Xeon(R) CPU E5-2650 v4 @ 2.20GHz
    * sockets: 2 
    * cores per socket: 12 
    * hyperthreading=yes(2) 
    * CPUs: 48
  * RAM: 
    * 256GB
  * Disks: 
    * Type: NVMe
    * Number: 3 connected in RAID0
  * Networking: 40Gb

* Loader server (2 loader servers are used):
  * Platform: physical
  * Architecture: PPC64
  * CPU: 
    * Model: POWER8 (raw), altivec supported, 2.0 (pvr 004d 0200)
    * sockets: 2 
    * cores per socket: 10
    * hyperthreading=yes(8) 
    * CPUs: 160
  * RAM: 
    * 256GB
  * Networking: 40Gb

Both scylla and loader machines were tuned using `perftune.py` in a `sq_split` mode.

*Scylla command*
```
scylla --options-file scylla.yaml --log-to-stdout 1 --memory 200G --default-log-level info --smp 46 --cpuset 1-11,25-35,12-23,36-47 --max-io-requests 600 --developer-mode false --collectd 0 --num-io-queues 46 --prometheus-address <public IP> --background-writer-scheduling-quota 0.5 --auto-adjust-flush-quota 1 --load-balance none --api-address <internal IP> --collectd-hostname node1
```

*Loaders description*
1. Stop scylla.
2. Clear the `data` and the `commitlog` directories.
3. Start scylla.
4. Run the test:
   * On each loader machine N, where N is from [0,1], run following commands:
      * Create the keyspace by running a short write test with `threads=1`:
```
scylla-tools-java/tools/bin/cassandra-stress write no-warmup n=1000 -node <internal IP> -rate threads=1  -mode native cql3 -pop seq=1..9000000 -schema keyspace=ks<N> replication\(strategy=NetworkTopologyStrategy, datacenter1=1\)
```
      * Run 14 loaders:
        * Loader K, where K is from [0, 13]: 
```
taskset -c <8 + 10*K>-<8 + 10*(K+1) - 1> scylla-tools-java/tools/bin/cassandra-stress write no-warmup n=9000000 -node <internal IP> -rate threads=200  -mode native cql3 -pop seq=<1 + 9000000*K>..<9000000*(K+1)> -schema keyspace=ks<N> replication\(strategy=NetworkTopologyStrategy, datacenter1=1\)
``` 

*The regression description*
I ran 3 iterations of the test above for the 2.1-rc3 branch at bdbbfe9 and then at its HEAD (3ab1c8abffdd2fe84945089ba0d363e357c65b55) and the throughput looks like this:
![image](https://user-images.githubusercontent.com/6103981/36916785-e33d5316-1e22-11e8-8795-86cc9f910204.png)

**Comment by glommer** [comment]
What were the actual numbers that you got ?

**Comment by vladzcloudius** [comment]
@glommer See the updated initial post (right above the graphs).

**Comment by vladzcloudius** [comment]
In addition to the regression in the 2.1 branch, there is a regression compared to an older tree: 4c920c98917dfa41ae500bee906e2bad9cb60fa7.
The regression of 3ab1c8a compared to that tree is ~8%.

**Comment by vladzcloudius** [comment]
By @glommer's suggestion I tested the same hashes with cache disabled and this is what I've got:
```
4c920c9 | bdbbfe9 | 3ab1c8ab
----------------------------
811096  | 773703  | 758554
804783  | 772623  | 755667
800362  | 770941  | 756019

Average
----------------------------
805414  | 772422 (-4%) | 756747 (-2%)
```
These results hit us that at least part of the regression can be related to changes in the cache - FYI @tgrabiec.
However another important input these numbers provide is that there is something else that is responsible for even greater regression.

**Comment by slivne** [comment]
@vladzcloudius - can you summarize the regression - was it 3% difference ?

**Comment by vladzcloudius** [comment]
@slivne No, it's ~9% difference as mentioned in my post from March 2 above.
